plugins/modem-manager/fu-plugin-modem-manager.c

```diff
@@ -243,8 +247,21 @@ fu_plugin_mm_device_removed_cb (MMManager *manager, MMObject *modem, FuPlugin *p
 	if (dev == NULL)
 		return;
 	g_debug ("removed modem: %s", mm_object_get_path (modem));
+
+#if MM_CHECK_VERSION(1,17,1)
+	{
+		MMModemFirmwareUpdateMethod dev_update_methods = fu_mm_device_get_update_methods(dev);
+		mbim_mm_object_path = g_strdup (object_path);
+		/* later removed the inhibited mbim device from the plugin */
+		if ((dev_update_methods & MM_MODEM_FIRMWARE_UPDATE_METHOD_MBIM_QDU) == 0) {
```

Comment:
Can you please explain why you're not calling `fu_plugin_cache_remove()` and `fu_plugin_device_remove()` when the update method is MBIM_QDU? What's the reason for that?

Reply:
I will call fu_plugin_cache_remove() and fu_plugin_device_remove() in `fu_plugin_update_attach ()` .

Because if I call them here, the download firmware progress bar will not respond during upgrade progress, and when I call them in `fu_plugin_update_attach ()`, the upgrade is normal.

Reply:
As said in a previous comment, in your case you don't really need update_detach and update_attach, as the modem does not change layout during the upgrade process.

Reply:
>This is the issue here. This modem is MBIM before, MBIM during upgrade, and MBIM after. It doesn't need update_detach() or update_attach(). The way to solves this is by making sure these steps don't run (or do nothing) in your case. I think the way you were doing it is by "mimicking" the need of these operations, while you should not need them at all. Could you see if you could rework that?

I think what you said is very reasonable.
Could this PR be merged first? And I will solve the problem of update_detach and update_attach in another PR. Thanks.

Reply:
> Could this PR be merged first? And I will solve the problem of update_detach and update_attach in another PR. Thanks.

I cannot decide that myself; if it was my decision, I would try to get it right before merging. If we merge a temporary solution before having the final one, we'll lose traction and interest in getting it right. What do @hughsie and @superm1 think?

Reply:
@aleksander0m I trust your judgement here dude.